### DISCUSSION TITLE
Table in a Modal

### DISCUSSION THREAD

**Discussion by moskoweb** [filamentphp/filament#5488] [Help]
Hello,

I need to put a table inside a modal, with information from a relationship, I wanted to know how I could create a table with 4 columns within a modal action?

**Comment by moskoweb** [comment]
There is no way to do it by the way, more than a month without an answer.

**Reply by danharrin** [reply]
There is, we're just not gonna write all the code for you. You can use custom modal content in HTML, as explained in the docs. https://filamentphp.com/docs/2.x/tables/actions#custom-content

**Reply by moskoweb** [reply]
I saw about the custom content, my idea would be how to use the Table class, used in resources in a modal, I've already tried creating a FilamentTable, but it didn't work, using the direct class, but it didn't work either.

My intention is to list the child elements of the relationship in modal.

**Reply by danharrin** [reply]
You can't embed a Filament table within a modal, but its absolutely possible to do it with Blade and HTML.

**Comment by waqasraza123** [comment]
was looking for the same functionality, but ended up creating a custom modal and custom table for the purpose. It would be nice if the filament tables can be embded in the modals.

**Comment by icaliman** [comment]
It is possible to create a **TableWidget** and embed it within a modal.


Example:
```php
Action::make('details')
    ->action(fn () => $this->record->someAction())
    ->modalContent(view('filament.resources.event.actions.details'))
```


File: `filament/resources/event/actions/details.blade.php`

```blade
@php
$widgetData = [
  'inModal' => true
];
@endphp

<x-filament::widgets :widgets="[\App\Filament\Widgets\DetailsWidget::class]" :columns="1" :data="$widgetData" />
```

File: `App\Filament\Widgets\DetailsWidget.php`

```php
class DetailsWidget extends TableWidget
{
    public $inModal = false;

    protected function getTableQuery(): Builder
    {
        return Activity::query();
    }

    protected function getTableActions(): array
    {
        return [
            Tables\Actions\Action::make('some-action'),
        ];
    }


    protected function getTableColumns(): array
    {
        return [
            Tables\Columns\ViewColumn::make('title')
                ->sortable()
                ->searchable(),
            Tables\Columns\TextColumn::make('created_at')
                ->sortable()
                ->since(),
        ];
    }

}
```

**Reply by phamduc1823** [reply]
\Illuminate\Database\Eloquent\Builder is it right

**Reply by Alex-S-Davies** [reply]
I think you have to use `<x-filament-widgets::widgets` instead of  `<x-filament::widgets` in your blade file now (works in Filament v3).

But this was a great time save, thanks very much

**Reply by lourbert** [reply]
@livewire(\App\Filament\Widgets\DetailsWidget::class)

**Comment by moskoweb** [comment]
I couldn't get it to work as you suggested in v2.0

**Reply by leandrocfe** [reply]
https://github.com/filamentphp/filament/discussions/8293#discussioncomment-6929362

**Comment by subhamsharmaa** [comment]
Here’s a neat trick to use a Filament table inside a modal:

1. Create a custom Livewire component and implement both HasForms and HasTable interfaces.
2. Define your table logic inside the component using the getTable() method. 
3.In your action, return a Blade view, and within that view, include your Livewire component like this:
<div>
    {{ $this->table }}
</div>
This approach works perfectly and allows you to render the Filament table inside a modal.

**Reply by mohammedsiraj** [reply]
can you describe it more

**Reply by subhamsharmaa** [reply]
1.Create an action
 `Tables\Actions\Action::make('view_payment')
                        ->modalContent(function($record){
                            return view('livewire.payment-layout',['record'=>$record]);
                        })
                        ->icon('heroicon-o-eye')
                        ->slideOver()`
                        
  2.  Create your livewire component `php artisan make:livewire YOURCOMPONENT ` (We can use livewire table and forms here)
  3. In the blade file of step 1: (livewire.payment-layout)
 ` <div> @livewire('your-class-name',['record'=>$record])</div>`

You can refer here for more detail: https://devnestle.com/how-to-add-filament-table-inside-modal-action/

**Reply by lukas-frey** [reply]
We made a package for this purpose if you still need it :-) https://github.com/GuavaCZ/filament-modal-relation-managers
